### ISSUE TITLE
Error on missing context variable

### ISSUE THREAD

**Issue by BebeSparkelSparkel** [jgm/doctemplates#18]
Is there a way to throw an error if there is a missing variable instead of defaulting to blank?

**Comment by BebeSparkelSparkel** [comment]
It would be useful to have a required syntax as a prefix to a variable name. Like `$$!name$$`

**Comment by jgm** [comment]
No, but you could write a function that walks the compiled template and compiles a list of variables, then use this to test against a Context.

**Comment by BebeSparkelSparkel** [comment]
I was thinking about that but it starts becoming more complex with the conditionals.
